webapp/channels/src/components/advanced_text_editor/use_post_box_indicator.tsx

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {DateTime} from 'luxon';
+import {useState, useEffect, useMemo} from 'react';
+import {useSelector} from 'react-redux';
+
+import {getDirectChannel} from 'mattermost-redux/selectors/entities/channels';
+import {isScheduledPostsEnabled} from 'mattermost-redux/selectors/entities/scheduled_posts';
+import {getTimezoneForUserProfile, getCurrentTimezone} from 'mattermost-redux/selectors/entities/timezone';
+import {getStatusForUserId, getUser, makeGetDisplayName} from 'mattermost-redux/selectors/entities/users';
+
+import Constants, {UserStatuses} from 'utils/constants';
+
+import type {GlobalState} from 'types/store';
+
+const DEFAULT_TIMEZONE = {
+    useAutomaticTimezone: true,
+    automaticTimezone: 'UTC',
+    manualTimezone: '',
+};
+
+const MINUTE = 1000 * 60;
+
+function useTimePostBoxIndicator(channelId: string) {
+    const getDisplayName = useMemo(makeGetDisplayName, []);
+
+    const teammateId = useSelector((state: GlobalState) => getDirectChannel(state, channelId)?.teammate_id || '');
+    const teammateDisplayName = useSelector((state: GlobalState) => (teammateId ? getDisplayName(state, teammateId) : ''));
+
+    const isDM = useSelector(
+        (state: GlobalState) => Boolean(getDirectChannel(state, channelId)?.teammate_id),
+    );
+
+    // Check if the teammate is in DND status
+    const isTeammateDND = useSelector((state: GlobalState) =>
+        (teammateId ? getStatusForUserId(state, teammateId) === UserStatuses.DND : false),
+    );
+
+    // Determine if the DND warning should be shown
+    const showDndWarning = isTeammateDND && isDM;
+
+    const [timestamp, setTimestamp] = useState(0);
+    const [showIt, setShowIt] = useState(false);
+
+    // get teammate timezone information
+    const teammateTimezone = useSelector(
+        (state: GlobalState) => {
+            if (!teammateId) {
+                return DEFAULT_TIMEZONE;
+            }
+
+            const teammate = getUser(state, teammateId);
+            return getTimezoneForUserProfile(teammate);
+        },
+        (a, b) =>
```

Comment:
Good solution for this here, but it makes me realize that `getTimezoneForUserProfile` should be a selector factory. It looks like it'll cause part of the user settings modal to re-render constantly when admins are editing another user's display settings

Reply:
solved here https://github.com/mattermost/mattermost/pull/29116/commits/654e2c9acd380899d6456df587ad439ef505830f

turned into selector @hmhealey